src/caret/movement.rs

```diff
@@ -175,4 +175,24 @@ impl Editor {
 
         None
     }
+
+    /// Get next WORD forward
+    /// "A WORD consists of a sequence of non-blank characters, separated with
+    /// whitespace.  An empty line is also considered to be a WORD."
+    pub fn _next_word_forward(&self, n: usize) -> Option<usize> {
```

Comment:
well this is for W which goes to the next WORD. w goes to next word.

we either need functions for both or make that one function do both.